ledger/src/tests.rs

```diff
@@ -2668,4 +2669,141 @@ mod valid_solutions {
             candidate_solutions.iter().skip(CurrentNetwork::MAX_SOLUTIONS).map(|s| s.id()).collect::<HashSet<_>>();
         assert_eq!(block_aborted_solution_ids, expected_aborted_solutions, "Aborted solutions do not match");
     }
+
+    #[test]
+    fn test_malicious_solution() {
+        // Initialize an RNG.
+        let rng = &mut TestRng::default();
+
+        // Initialize the test environment.
+        let crate::test_helpers::TestEnv { ledger, private_key, address, .. } =
+            crate::test_helpers::sample_test_env(rng);
+
+        // Retrieve the puzzle parameters.
+        let puzzle = ledger.puzzle();
+        let latest_epoch_hash = ledger.latest_epoch_hash().unwrap();
+        let minimum_proof_target = ledger.latest_proof_target();
+
+        // Initialize a valid solution object.
+        let mut valid_solution = None;
+        while valid_solution.is_none() {
+            let solution = puzzle.prove(latest_epoch_hash, address, rng.gen(), None).unwrap();
+            if puzzle.get_proof_target(&solution).unwrap() >= minimum_proof_target {
+                valid_solution = Some(solution);
+            }
+        }
+        // Unwrap the valid solution.
+        let valid_solution = valid_solution.unwrap();
+
+        // Construct a malicious solution with a different target.
+        let malicious_solution =
+            Solution::new(*valid_solution.partial_solution(), valid_solution.target().saturating_add(1));
```

Comment:
Could we use `minimum_proof_target - 1` instead?

Reply:
Resolved with - [3adb1e2](https://github.com/AleoNet/snarkVM/pull/2520/commits/3adb1e26db563692d811dcdf40994f2e55975a76)